added post-request, installed immer

iraskiba opened the pull request:
added post-request to the edit-button from the tasks-page, installed immer and created a store for storing pagination and sorting for customers-page

[Removed immer,reduced imports in some components,changed storage and added reset function for future sorting of all customers](https://github.com/iraskiba/CRMinter/pull/9/commits/284ea4c4092c768a36f3ac1c0ace637a5b721a80)

 [removed use-sync-external-store,react-use; installed auto-zustand-selectors-hook,](https://github.com/iraskiba/CRMinter/pull/9/commits/1dc04290a4caf440e36fe392c5498af8cc32b1b8)
changed saving of post query result

[BASE_URL constant was created in the chared folder,](https://github.com/iraskiba/CRMinter/pull/9/commits/dda4e95d565bcd947fd4ed6840792564de57437a)
created interceptors in the lib folder to check the request and respond to errors for different url's,
 the query was moved to the api folder in the customers directory,
 and a folder for table columns was created 

[fix:api folders,totalCount of customers,onrow](https://github.com/iraskiba/CRMinter/pull/9/commits/3c8ad4f649df5fe40885493518d59effe582a4c1)
api folders were created in customers, deals, tasks.
added query interceptors to the functions of these folders.
The totalCount of customers will be stored in the table storage. onRow function has been changed.

added event bus. added notification on pressing the ‘Save Task’-button in the modal window

iraskiba committed:
added post request to the edith button from the task page, installed immer and created a store for storing pagination and sorting

Khomko-Dima commented on package.json:
```diff
@@ -22,13 +22,16 @@
     "axios": "^1.7.7",
     "customize-cra": "^1.0.0",
     "dayjs": "^1.11.13",
+    "immer": "^10.1.1",
```
а зачем immer?

Khomko-Dima commented on src/pages/customers/ui/customers.tsx:
```diff
@@ -7,9 +7,10 @@ import {
+import { useEffect } from 'react'
 import { useQuery } from '@tanstack/react-query'
 import { useNavigate } from 'react-router-dom'
+import usePaginationStore from '@pages/customers/model/store.ts'
```
сократи импорт

Khomko-Dima commented on src/pages/customers/model/store.ts:
```diff
@@ -0,0 +1,50 @@
+import { immer } from 'zustand/middleware/immer'
+
+type State = {
+  page: number
```
заверни все данные в объект params
сможешь в запрос сразу объект передавать а не доставать все параметры по 1

Khomko-Dima commented on src/pages/tasks/ui/edit-task-modal.tsx:
```diff
@@ -1,60 +1,73 @@
+}
+
 const EditTaskModal: FC<TaskModalProps> = ({ visible, onClose }) => {
+  const postTask = async (task: TaskFormValues) => {
```
вынеси из компонета

iraskiba committed:
Removed immer,reduced imports in some components,changed storage and added reset function for future sorting of all customers

Khomko-Dima commented on package.json:
```diff
@@ -27,8 +27,10 @@
     "react-hook-form": "^7.53.0",
     "react-is": "^18.3.1",
     "react-router-dom": "^6.26.1",
+    "react-use": "^17.5.1",
```
удали

Khomko-Dima commented on package.json:
```diff
@@ -27,8 +27,10 @@
+    "react-use": "^17.5.1",
     "sass": "^1.77.8",
     "typescript-eslint": "^8.1.0",
+    "use-sync-external-store": "^1.2.2",
```
тоже удали

Khomko-Dima commented on src/pages/tasks/model/api.tsx:
```diff
@@ -0,0 +1,11 @@
+  description: string
+}
+export const postTask = async (task: TaskFormValues) => {
+  const response = await axios.post('http://localhost:3001/tasks', task)
```
const { data } = await..
return data

Khomko-Dima replied:
const {data} = await axios.post('http://localhost:3001/tasks', task)
return data

iraskiba committed:
removed use-sync-external-store,react-use; installed auto-zustand-selectors-hook,
changed saving of post query result

iraskiba committed:
BASE_URL constant was created in the chared folder,
created interceptors in the lib folder to check the request and respond to errors for different url's,
 the query was moved to the api folder in the customers directory,
 and a folder for table columns was created

Khomko-Dima commented on src/pages/customers/index.ts:
```diff
@@ -1,2 +1,5 @@
 import Customers from './ui/customers.tsx'
-export default Customers
+import usePaginationStore from './model/store.ts'
+import CustomersColumns from './ui/customers-columns.tsx'
```
а зачем экспортировать колонки?

Khomko-Dima commented on src/pages/customers/ui/customers-columns.tsx:
```diff
@@ -0,0 +1,52 @@
+  avatarProps?: AvatarProps
+}
+
+const CustomersColumns = (): ColumnsType<Customer> => [
```
Зачем тут функция если можно просто массив?

Khomko-Dima commented on src/pages/customers/ui/customers.tsx:
```diff
@@ -1,96 +1,39 @@
-  return response.data
-}
+import { usePaginationStore } from '@pages/customers'
+import { Customer, fetchCustomers } from '@pages/customers/model/api.tsx'
```
запросы (api) должны быть в папке api

Khomko-Dima commented on src/pages/customers/ui/customers.tsx:
```diff
@@ -1,96 +1,39 @@
+
+  useEffect(() => {
+    if (data) {
+      setParams({ totalCount: data.totalCount })
```
зачем хранить в store?
totalCount  только в таблице лежит, можно там и использовать

Khomko-Dima commented on src/pages/customers/ui/customers.tsx:
```diff
@@ -105,18 +48,20 @@ const Customers = () => {
-        onRow={(record) => ({
-          onClick: () => handleClick(record),
-        })}
+        onRow={(record) => {
```
предыдущий вариант был лучше

iraskiba committed:
fix:api folders,totalCount of customers,onrow

api folders were created in customers, deals, tasks.
added query interceptors to the functions of these folders.
The totalCount of customers will be stored in the table storage. onRow function has been changed.

iraskiba committed:
fead: event bus

added event bus. added notification on pressing the ‘Save Task’-button in the modal window

Khomko-Dima commented on src/pages/tasks/model/api.tsx:
```diff
@@ -0,0 +1,11 @@
+  description: string
+}
+export const postTask = async (task: TaskFormValues) => {
+  const response = await axios.post('http://localhost:3001/tasks', task)
```
а ещё сделай так чтоб запросы были вида await axios.post('/tasks', task)
можно сделать через https://axios-http.com/docs/instance
и можно настроить env.proccess

iraskiba committed:
created axios component for instance

iraskiba committed:
fix:fixed request port

iraskiba committed:
added function for creating interceptors

added  the function for creating interceptors   and called in the app component

iraskiba committed:
removed rule about any type
